src/test/java/org/openrewrite/java/testing/hamcrest/RemoveIsMatcherTest.java

```diff
@@ -68,4 +68,50 @@ void testEquals() {
             """));
     }
 
+    @Test
+    void isNotCalledInAssertThat() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.junit.jupiter.api.Test;
+              import static org.hamcrest.Matchers.is;
+              import static org.hamcrest.Matchers.equalTo;
+              
+              class ATest {
+                  @Test
+                  void testMethod() {
+                      String str1 = "Hello world!";
+                      Matcher<String> x = is(equalTo(str1));
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void isNotDirectlyInAssertThat() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.junit.jupiter.api.Test;
+              import static org.hamcrest.MatcherAssert.assertThat;
+              import static org.hamcrest.Matchers.is;
+              import static org.hamcrest.Matchers.equalTo;
+              
+              class ATest {
+                  @Test
+                  void testMethod() {
+                      String str1 = "Hello world!";
+                      String str2 = "Hello world!";
+                      assertThat(str1, contains(is(equalTo(str2))));
+                  }
+              }
+              """
```

Comment:
When using `not` we might actually want to execute the recipe, because there's still a wrapping `assertThat`.
```suggestion
              """,
            """
              import org.junit.jupiter.api.Test;
              import static org.hamcrest.MatcherAssert.assertThat;
              import static org.hamcrest.Matchers.not;
              import static org.hamcrest.Matchers.equalTo;
              
              class ATest {
                  @Test
                  void testMethod() {
                      String str1 = "Hello world!";
                      String str2 = "Hello world!";
                      assertThat(str1, not(equalTo(str2)));
                  }
              }
              """
```

Reply:
I see, so `is()` should be replaced when it is nested inside certain function calls that are inside `assertThat()`? Which methods like this do we want the recipe to execute on besides not?

Reply:
It's not specific to `not` or any other matcher; we want to remove `is(..)` whenever that's used within an `assertThat` method call (however deep), but not when used outside an `assertThat` call. 

We add this limitation to this recipe because hamcrest matchers are also sometimes used in different frameworks such as restassured; our migration recipes do not cover those cases, and we can't be certain it's safe or desirable to remove `is` method invocations there.